tracer/src/Datadog.Tracer.Native/debugger_probes_tracker.h

```diff
@@ -28,6 +29,9 @@ namespace debugger
         // Holds indices that were previously used and freed upon probes removal.
         // the `_probeMetadataMapMutex` is used for syncing
         inline static std::queue<int> _freeProbeIndices{};
+        // Holds incremental number that is uniquely given for each and every instrumentation instance. If a probe is added/removed from
+        // a specific method, then this method is going to get a new number.
+        inline static std::atomic<int> _nextInstrumentationSequence{0};
```

Comment:
Same comment as I made in managed code about the use of the word `sequence`, I'd recommend that if we decide to rename, we should rename across both the entire PR, for both the managed and native parts.